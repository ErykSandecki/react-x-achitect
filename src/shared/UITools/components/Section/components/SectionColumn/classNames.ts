export const className = 'SectionColumn';

export const classNames = {
  [className]: className,
  content: `${className}__content`,
  buttons: `${className}__buttons`,
};
