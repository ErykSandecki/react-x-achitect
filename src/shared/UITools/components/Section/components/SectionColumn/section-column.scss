.SectionColumn {
  align-items: center;
  column-gap: 8px;
  display: flex;

  &__content {
    width: calc(100% - 24px - 8px);
  }

  &__buttons {
    align-items: center;
    column-gap: 2.5px;
    display: flex;
    flex-grow: 1;
    justify-content: center;
  }
}
