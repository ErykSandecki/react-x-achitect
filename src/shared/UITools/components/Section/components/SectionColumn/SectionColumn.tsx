import { FC, ReactNode } from 'react';

// components
import Box from '../../../../../UI/components/Box/Box';

// hooks
import { useTheme } from 'hooks';

// others
import { className, classNames } from './classNames';

// styles
import styles from './section-column.scss';

// types
import { E2EAttribute } from 'types';

export type TSectionColumnProps = {
  buttonsIcon?: Array<ReactNode>;
  children: ReactNode;
};

export const SectionColumn: FC<TSectionColumnProps> = ({
  buttonsIcon = [],
  children,
}) => {
  const { classNamesWithTheme, cx } = useTheme(classNames, styles);

  return (
    <Box
      classes={{ className: cx(classNamesWithTheme[className]) }}
      e2eAttribute={E2EAttribute.section}
    >
      <div className={cx(classNamesWithTheme.content)}>{children}</div>
      <div className={cx(classNamesWithTheme.buttons)}>
        {buttonsIcon.map((buttonIcon) => buttonIcon)}
      </div>
    </Box>
  );
};

export default SectionColumn;
