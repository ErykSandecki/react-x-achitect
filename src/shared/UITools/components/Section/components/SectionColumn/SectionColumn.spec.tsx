import { render } from '@testing-library/react';

// components
import SectionColumn from './SectionColumn';

describe('SectionColumn snapshots', () => {
  it('should render SectionColumn', () => {
    // before
    const { asFragment } = render(<SectionColumn>children</SectionColumn>);

    // result
    expect(asFragment()).toMatchSnapshot();
  });
});
