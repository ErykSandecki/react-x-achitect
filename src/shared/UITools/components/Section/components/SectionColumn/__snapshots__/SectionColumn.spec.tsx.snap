// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SectionColumn snapshots should render SectionColumn 1`] = `
<DocumentFragment>
  <div
    class="css-q0izro SectionColumn SectionColumn--light"
    data-test-section=""
  >
    <div
      class="SectionColumn__content SectionColumn__content--light"
    >
      children
    </div>
    <div
      class="SectionColumn__buttons SectionColumn__buttons--light"
    />
  </div>
</DocumentFragment>
`;
